src/config/validation.rs

```diff
@@ -7,10 +7,37 @@ pub fn check_shape(config: &Config) -> Result<(), Vec<String>> {
     if config.sources.is_empty() {
         errors.push("No sources defined in the config.".to_owned());
     }
+
     if config.sinks.is_empty() {
         errors.push("No sinks defined in the config.".to_owned());
     }
 
+    // Helper for below
+    fn tagged<'a>(
+        tag: &'static str,
+        iter: impl Iterator<Item = &'a String>,
+    ) -> impl Iterator<Item = (String, &'a String)> {
+        iter.map(move |x| (tag.to_owned(), x))
+    }
+
+    // Check for non-unique names across sources, sinks, and transforms
+    let mut name_uses = HashMap::<&str, Vec<String>>::new();
```

Comment:
Could this be `HashMap::<&str, Vec<&'static str>>` to avoid cloning the tag?

Reply:
Yes it could!